src/infra/core/src/compact_service_impl.rs

```diff
@@ -0,0 +1,417 @@
+                            || batch_records + current_records > max_slice_records
+                        {
+                            match data_slice_batch_info.data_slices_batch.len().cmp(&1) {
+                                Ordering::Equal => data_slice_batches
+                                    .push(DataSliceBatch::SingleBlock(current_hash.unwrap())),
+                                Ordering::Greater => {
+                                    data_slice_batches.push(DataSliceBatch::CompactedBatch(
+                                        Box::new(data_slice_batch_info.clone()),
+                                    ));
+                                }
+                                _ => (),
+                            };
+                            if !data_slice_batch_info.data_slices_batch.is_empty() {
+                                // Reset values for next batch
+                                data_slice_batch_info = DataSliceBatchInfo::default();
+                                data_slice_batch_info.upper_bound.end_offset =
+                                    output_slice.offset_interval.end;
+                            }
+
+                            data_slice_batch_info.data_slices_batch = vec![data_slice_url];
+                            batch_size = output_slice.size;
+                            batch_records = current_records;
+                        } else {
+                            data_slice_batch_info.data_slices_batch.push(data_slice_url);
+                            batch_size += output_slice.size;
+                            batch_records += current_records;
+                        }
+
+                        // Set lower bound values
+                        data_slice_batch_info.lower_bound.prev_checkpoint =
+                            add_data_event.prev_checkpoint;
+                        data_slice_batch_info.lower_bound.prev_offset = add_data_event.prev_offset;
+                        data_slice_batch_info.lower_bound.start_offset =
+                            output_slice.offset_interval.start;
+                        current_hash = Some(block_hash);
+
+                        // Set upper bound values
+                        if data_slice_batch_info.upper_bound.new_checkpoint.is_none() {
+                            data_slice_batch_info.upper_bound.new_checkpoint =
+                                add_data_event.new_checkpoint;
+                        }
+                        if data_slice_batch_info.upper_bound.new_source_state.is_none() {
+                            data_slice_batch_info.upper_bound.new_source_state =
+                                add_data_event.new_source_state;
+                        }
+                    }
+                }
+                MetadataEvent::SetVocab(set_vocab_event) => {
+                    offset_column = set_vocab_event.offset_column;
+                }
+                _ => continue,
```

Comment:
Better to use `_ => ()` for a no-op, as `continue` will interrupt the flow of the loop which is not needed here.

Reply:
Changed